src/python/WMComponent/RucioInjector/Database/MySQL/GetCompletedBlocks.py

```diff
@@ -86,21 +85,29 @@ def format(self, result):
         # NOTE: We need to rename all the keys to follow the cammelCase standard. And also to comply
         #       with the key names as expected from the rest of the already existing python code
         keyMap = {'blockname': 'blockName',
-                  'name': 'workflowName',
+                  'name': 'workflowNames',
                   'pnn': 'location',
                   'site': 'sites',
                   'path': 'dataset',
                   'create_time': 'blockCreateTime'}
 
-        dictResults = DBFormatter.formatDict(self, result)
-        for record in dictResults:
-            for dbKey, pyKey in keyMap.items():
-                if dbKey == 'site':
-                    sites = record.pop(dbKey)
-                    record[pyKey] = set()
-                    record[pyKey].add(sites)
-                else:
-                    record[pyKey] = record.pop(dbKey)
+        listResults = DBFormatter.formatDict(self, result)
+        dictResults = {}
+        for record in listResults:
+            # Populates results dict and adds all workflows of the same block to a single record
+            blockname = record['blockname']
+            if blockname in dictResults:
```

Comment:
In general the DAO must return the list of results as it is from the DABASE and no extra data reformatting should be needed. It seems to me that if this extra aggregation by block name here in the python code is needed, then something is not quite working as we expect in the `GROUP BY` clause in the SQL query.

Reply:
We can do this aggregation outside of the DAO. Also, we may be able to modify the query so it ignores blocks with blocks that are not completed, but then that's probably redundant with the deletableWorkflows query

Reply:
Thanks @germanfgv!  We largely discussed the problem an tested multiple times with you, and we now know in details what is going on. I'd suggest  we  continue as it is. And also since this DAO is used at a single place in the WMCore code, my previous remark that we should avoid the aggregation in the DAO and leave it at the component would only make the code at the component more unreadable. So lets proceed with what we already have. Just put the comments, as we discussed during our meeting, mentioning what are the background condition we rely on in order for that logic to work. So that we do remember those little but extremely important details. Thanks!